source/agora/crypto/Schnorr.d

```diff
@@ -199,10 +237,29 @@ public Signature sign (T) (
       To get `c`, need to precommit `R`
      */
     // Compute the challenge and reduce the hash to a scalar
-    Scalar c = hashFull(const(Message!T)(X, R, data));
+    const Scalar c = hashFull(const(Message!T)(X, R, data));
+    return sign(x, R, r, c);
+}
+
+/// sign with prepared message hash `c` (used for multisig with threshold)
+public Signature sign (in Scalar x, in Point R, in Scalar r, in Scalar c)
+    nothrow @nogc @trusted
```

Comment:
Why `@trusted` ?

Reply:
Updated this a a couple of other functions.